ethereum/contracts/IncentivizedInboundChannel.sol

```diff
@@ -54,19 +58,25 @@ contract IncentivizedInboundChannel is InboundChannel {
             "invalid commitment"
         );
 
-        // Require there is enough gas to play all messages
-        require(
-            gasleft() >= _messages.length * MAX_GAS_PER_MESSAGE,
-            "insufficient gas for delivery of all messages"
-        );
-
         // Require all payloads are smaller than max_payload_size
         for (uint256 i = 0; i < _messages.length; i++) {
             require(
                 _messages[i].payload.length <= MAX_PAYLOAD_BYTE_SIZE,
                 "message payload bytesize exceeds maximum payload size"
             );
         }
+
+        // Require there is enough gas to play all messages
+        // TODO: Test/review further.
+        uint256 maxMessageProcessingCost =
+            _messages.length * MAX_GAS_PER_MESSAGE;
+        require(
+            gasleft() >=
+                maxMessageProcessingCost +
+                    POST_PROCESSING_LEFTOVER_GAS_REQUIRED,
+            "insufficient gas for delivery of all messages"
+        );
+
```

Comment:
Do we even need those gas checks? When we execute the calls, we limit the gas, which might be enough?

Reply:
yeh i think you're right

Reply:
Should we remove the gas checks this PR or make an issue for a follow-up PR?

Reply:
I created issue https://github.com/Snowfork/polkadot-ethereum/issues/268 for investigating these potentially redundant gas checks. I can address it in this PR if it's blocking.